src/centrifuge.ts

```diff
@@ -1261,6 +1250,9 @@ export class Centrifuge extends (EventEmitter as new () => TypedEventEmitter<Cli
       const transport = this._transport;
       this._transport = null;
       transport.close(); // Close only after setting this._transport to null to avoid recursion when calling transport close().
+      // Need to mark as closed here, because connect call may be sync called after disconnect,
+      // transport onClose callback will not be called yet
+      this._transportIsOpen = false;
```

Comment:
I think this should be moved on top of `disconnect`. Otherwise existing test cases result into `already subscribed` error on server side in `subscribe and presence` test case (because undesired subscribe frame was sent upon disconnecting).

It happens because disconnect moves Subscription to Subscribing state (in `_clearConnectedState` method) . `_setSubscribing` of Subscription is called, then `_subscribe` is called. Since transport is still open client tries to send subscribe frame to server and succeeds.

Moving this on top (after `if (this._isDisconnected()) {` check) solved the problem for me.

Reply:
Found the reason it been sent, it's called after this call
<img width="706" alt="image" src="https://github.com/centrifugal/centrifuge-js/assets/13537179/2bede2c0-5660-47a9-aded-cc58473c69ae">

I have added a check inside `_setSubscribing`, for the reason of changing and calling _subscribe, only for `subscribingCodes.subscribeCalled`, I think its lit bit more clear condition, that will not break in case of some refactoring about transports, what do you think about it?

Reply:
Added a comment for a newly added check.